test/unit/mdc-dialog/foundation.test.js

```diff
@@ -0,0 +1,240 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and * limitations under the License.
+ */
+
+import {assert} from 'chai';
+import td from 'testdouble';
+
+import {setupFoundationTest} from '../helpers/setup';
+import {verifyDefaultAdapter} from '../helpers/foundation';
+
+import MDCDialogFoundation from '../../../packages/mdc-dialog/foundation';
+
+suite('MDCDialogFoundation');
+
+test('exports cssClasses', () => {
+  assert.isOk('cssClasses' in MDCDialogFoundation);
+});
+
+test('default adapter returns a complete adapter implementation', () => {
+  verifyDefaultAdapter(MDCDialogFoundation, [
+    'setBackgroundAriaHidden', 'setDialogAriaHidden', 'hasClass', 'addClass',
+    'removeClass', 'addScrollLockClass', 'removeScrollLockClass',
+    'registerInteractionHandler', 'deregisterInteractionHandler',
+    'registerDialogInteractionHandler', 'deregisterDialogInteractionHandler',
+    'registerDocumentKeydownHandler', 'deregisterDocumentKeydownHandler',
+    'registerAcceptHandler', 'deregisterAcceptHandler',
+    'registerCancelHandler', 'deregisterCancelHandler',
+    'registerFocusTrappingHandler', 'deregisterFocusTrappingHandler', 'setDefaultFocus',
+    'getFocusableElements', 'saveElementTabState', 'restoreElementTabState',
+    'makeElementUntabbable', 'setAttribute', 'acceptAction', 'cancelAction',
+    'getFocusedElement', 'setFocusedElement', 'numFocusableElements', 'resetDialogFocus',
+  ]);
+});
+
+function setupTest() {
+  return setupFoundationTest(MDCDialogFoundation);
+}
+
+test('#isOpen returns true when the dialog is open', () => {
+  const {foundation} = setupTest();
+
+  foundation.open();
+  assert.isOk(foundation.isOpen());
+});
+
+test('#isOpen returns false when the dialog is closed', () => {
+  const {foundation} = setupTest();
+
+  foundation.close();
+  assert.isNotOk(foundation.isOpen());
+});
+
+test('#open registers all events registered within open()', () => {
+  const {foundation, mockAdapter} = setupTest();
```

Comment:
Instead of having this line for most tests, would it be better to use a `testFoundation` helper like [here](https://github.com/material-components/material-components-web/blob/master/test/unit/mdc-menu/simple.foundation.test.js#L135)?